File: src/test/nodejs/typeorm/data-test/src/entity/AllTypes.ts
// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { Entity, PrimaryColumn, Column } from "typeorm"

@Entity("all_types")
export class AllTypes {

  @PrimaryColumn({primaryKeyConstraintName: 'PK_all_types', type: 'bigint'})
  col_bigint: number

  @Column({type: 'bool'})
  col_bool: boolean

  @Column({type: 'bytea'})
  col_bytea: Buffer

  @Column({type: 'float8'})
  col_float8: number

  @Column({type: 'int'})
  col_int: number

  @Column({type: 'numeric'})
  col_numeric: number

  @Column({type: 'timestamptz'})
  col_timestamptz: Date

  @Column({type: 'date'})
  col_date: string

  @Column({type: 'varchar', length: 100})
  col_varchar: string
}

File: src/test/python/sqlalchemy/all_types.py
from sqlalchemy import Column, Integer, String, Boolean, LargeBinary, Float,\
  Numeric, DateTime, Date
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.orm import registry

mapper_registry = registry()
Base = mapper_registry.generate_base()


class AllTypes(Base):
  __tablename__ = "all_types"

  col_bigint = Column(Integer, primary_key=True)
  col_bool = Column(Boolean)
  col_bytea = Column(LargeBinary)
  col_float8 = Column(Float)
  col_int = Column(Integer)
  col_numeric = Column(Numeric)
  col_timestamptz = Column(DateTime(timezone=True))
  col_date = Column(Date)
  col_varchar = Column(String)
  col_jsonb = Column(JSONB)

  def __repr__(self):
    return f"AllTypes(" \
           f"col_bigint=     {self.col_bigint!r}," \
           f"col_bool=       {self.col_bool!r}," \
           f"col_bytea=      {self.col_bytea!r}" \
           f"col_float8=     {self.col_float8!r}" \
           f"col_int=        {self.col_int!r}" \
           f"col_numeric=    {self.col_numeric!r}" \
           f"col_timestamptz={self.col_timestamptz!r}" \
           f"col_date=       {self.col_date!r}" \
           f"col_varchar=    {self.col_varchar!r}" \
           f"col_jsonb=      {self.col_jsonb!r}" \
           f")"
